### DISCUSSION TITLE
After ctrl+C my ESP32 S2 immediately starts running, so I can’t use AMPY to upload changes

### DISCUSSION THREAD

**Discussion by Quadrangle** [micropython/micropython#16429] [Using MicroPython]
Sorry for what has to be a dumb question.

Using Thonny, the STOP button does not work.  Using PyCharm, similar problem.  Using AMPY in the terminal window of VS Code, same problem.  I am unable to get program my ESP32 to stop so I can upload changes.

I’ve tried ctrl+C, as well as ctrl+A, B and D.  

I’ve tried things like stopping the program programmatically using the button on pin 1 (labelled “BOOT”) to disable my homegrown web server and notify browser to stop polling, and then exit main loop.

There are countless “Blinky” programs on the web and they all have the same problem for me.

I’ve tried soft and hard resets, all to no avail.

What am I doing wrong?

**Comment by peterhinch** [comment]
Code placed in `main.py` automatically runs on boot. When testing code designed to run on boot it's usually best to write `main.py` as follows
```py
import time
time.sleep(3)
import my_actual_code
```
The `sleep` gives time to break in with `ctrl-c` immediately after a boot.

If you can't access your device do a flash erase followed by re-installing firmware.
